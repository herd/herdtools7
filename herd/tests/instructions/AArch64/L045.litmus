AArch64 L045
Orig=DMB.SYdRW Rfe Amo.Swp PodWR Amo.LdAdd Rfe
{
0:X0=x; 0:X3=y;
1:X0=y; 1:X3=x;
}
 P0          | P1               ;
 LDR W1,[X0] | MOV W2,#2        ;
 DMB SY      | SWP W2,W1,[X0]   ;
 MOV W2,#1   | MOV W5,#1        ;
 STR W2,[X3] | LDADD W5,W4,[X3] ;
exists ([y]=2 /\ 0:X1=1 /\ 1:X1=1 /\ 1:X4=0)
