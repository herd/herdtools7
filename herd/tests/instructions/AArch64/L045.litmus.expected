Test L045 Allowed
States 4
0:X1=0; 1:X1=0; 1:X4=0; [y]=1;
0:X1=0; 1:X1=1; 1:X4=0; [y]=2;
0:X1=1; 1:X1=0; 1:X4=0; [y]=1;
0:X1=1; 1:X1=1; 1:X4=0; [y]=2;
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists ([y]=2 /\ 0:X1=1 /\ 1:X1=1 /\ 1:X4=0)
Observation L045 Sometimes 1 3
Hash=b554dbbcf6abfbb3161a8c7ca301e99b

